bin/provision-instances

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/python3
+#
+# This script runs a full instance provision over all non-ephemeral instances
+# on OCIM. Running this script requires Django to be initialized - the easiest
+# way to do this is by evaluating the script from within the Django shell by
+# running:
+#
+#   > eval(open('bin/provision-instances').open())
+#
+# The provision process happens in "waves", where all instances that fail to
+# provision in a given wave are recorded and reprovisioned during the next wave.
+#
+# This provision script is designed to withstand restarts of either OCIM or
+# this script itself; it accomplishes this by always saving and loading the
+# state of the provision process to/from `ocim_update_waves.ini`.
```

Comment:
@bdero 

>  if we make this a management command that uses the database for storage, should we also consider having this task be launchable via a web interface instead, and make sure OCIM will resume it in a worker post-restart?

Exactly - I think eventually this should be a standard feature of OpenCraft IM, and making it launchable from the web interface, with automated resumes in workers post-restart seem great. The more automated and transparent we make this, the easier it will make our lives, and the more likely we are to be using it and be keeping our instances up to date. 

We might even find ways to automate the triggering of these reprovisionings along the way, to make it even more transparent (like we did for deprovisioning - we avoided having to care about it by automating deprovisioning of old VMs). Here for example, we could possibly detect when there has been code changes, either to the edX repositories the instances point to, or to the code of OpenCraft IM.

And yes, again, this can be done progressively/iteratively over the next tickets that need to reprovision instances.